Source/FieldSolver/FiniteDifferenceSolver/ApplySilverMuellerBoundary.cpp

```diff
@@ -0,0 +1,157 @@
+/* Copyright 2021
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "Utils/WarpXAlgorithmSelection.H"
+#include "FiniteDifferenceSolver.H"
+#include "Utils/WarpXConst.H"
+#include <AMReX_Gpu.H>
+
+using namespace amrex;
+
+/**
+ * \brief Update the E field at the boundary, using the Silver-Mueller condition
+ */
+void FiniteDifferenceSolver::ApplySilverMuellerBoundary (
+    std::array< std::unique_ptr<amrex::MultiFab>, 3 >& Efield,
```

Comment:
would Efield have a const before it then?

Reply:
Good point! Actually I removed the `const` for `B` now, as it was confusing.

Reply:
I think the `Efield` is only read, so array, pointer and MultiFab are `const`.

The `Bfield` is manipulated, so only the array and pointer are const, not the MultiFab